File: SemanticSearch/PHP/Web/Plain/src/Storage/VectorFactory.php
<?php
require_once __DIR__ . '/VectorAdapter.php';
require_once __DIR__ . '/Adapters/ChromaDB.php';
require_once __DIR__ . '/Adapters/Pinecone.php';
require_once __DIR__ . '/Adapters/PgVector.php';

class VectorFactory {
    public static function create(): VectorStoreAdapter {
        $driver = getenv('VECTOR_DRIVER') ?: 'chromadb';
        return match ($driver) {
            'pinecone' => new Pinecone(),
            'pgvector' => new PgVector(),
            default => new ChromaDB(),
        };
    }
}

File: SemanticSearch/Python/Web/Plain/src/storage/vector_factory.py
import os
from storage.vector_adapter import VectorStoreAdapter
from storage.adapters.chromadb import ChromaDB
from storage.adapters.pinecone import Pinecone
from storage.adapters.pgvector import PgVector


def create_vector_store() -> VectorStoreAdapter:
    driver = os.getenv("VECTOR_DRIVER", "chromadb")
    if driver == "pinecone":
        return Pinecone()
    if driver == "pgvector":
        return PgVector()
    return ChromaDB()